third_party/nvfuser/csrc/compute_at_map.cpp

```diff
@@ -1106,6 +1127,9 @@ void ComputeAtMap::buildConcreteIds() {
         "Cannot compute concrete id of empty set.");
     auto first_id = disjoint_set_shared_ptr->vector().front();
     auto concrete_id = computeConcreteId(first_id, IdMappingMode::PERMISSIVE);
+    if (concrete_id_map_.find(concrete_id) != concrete_id_map_.end()) {
```

Comment:
`computeConcreteId` is also modifed to use `concrete_id_map_`. Is the change here also necessary? Same for the `ALMOSTEXACT` and `LOOP` maps.

I tried commenting out them and didn't see any error with `FusionScatter2DZerosSelfTvFusion_CUDA`.

Reply:
If we don't change `ALMOSTEXACT`. The vectorize pass will give some failure.  For `LOOP` map, we need to generate loop according to the `tv_index`, not `tv_output` or `tv_input`. So, we need to change it. I have added test case `FusionScatterNoParallism_CUDA` and delete some unused logic to show this.